{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Count reads from the merged files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os, glob, re, sys, gzip, time\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq \n",
    "from pathlib import Path\n",
    "\n",
    "\n",
    "from ngs_utils import * "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "RUN_TO_COUNT=2\n",
    "\n",
    "target_path = '../Data/designed_variants.csv'\n",
    "path_prefix = f'../Data/run{RUN_TO_COUNT}/Parsed_data/'\n",
    "\n",
    "\n",
    "merge_path=f'{path_prefix}/merged/'\n",
    "output_path = f'{path_prefix}/library/'\n",
    "\n",
    "Path(output_path).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "T21_F = 'aaggtcatgattaca'.upper() #forward primer\n",
    "T21_R = 'caagcagctaccgca'.upper() #reverse primer\n",
    "MIN_QSCORE_THRESHOLD = 20"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Count reads from all the lanes (This takes a few hours to run)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../Data/run1/Parsed_data//library/sorted\n",
      "../Data/run1/Parsed_data//library/lowQ\n",
      "../Data/run1/Parsed_data//library/discarded\n",
      "../Data/run1/Parsed_data//merged/S015_EK269_GAS1_p1_rep1c_plasmid-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S015_EK269_GAS1_p1_rep1c_plasmid-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S015_EK269_GAS1_p1_rep1c_plasmid-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S015_EK269_GAS1_p1_rep1c_plasmid-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S015_EK269_GAS1_p1_rep1c_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S015_EK269_GAS1_p1_rep1c_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S015_EK269_GAS1_p1_rep1c_plasmid.txt\n",
      "../Data/run1/Parsed_data//merged/S042_EK269_GAS1_v3_rep1b_virus-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S042_EK269_GAS1_v3_rep1b_virus-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S042_EK269_GAS1_v3_rep1b_virus-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S042_EK269_GAS1_v3_rep1b_virus-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S042_EK269_GAS1_v3_rep1b_virus.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S042_EK269_GAS1_v3_rep1b_virus.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S042_EK269_GAS1_v3_rep1b_virus.txt\n",
      "../Data/run1/Parsed_data//merged/S045_EK269_GAS1_v4_rep2a_virus-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S045_EK269_GAS1_v4_rep2a_virus-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S045_EK269_GAS1_v4_rep2a_virus-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S045_EK269_GAS1_v4_rep2a_virus-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S045_EK269_GAS1_v4_rep2a_virus.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S045_EK269_GAS1_v4_rep2a_virus.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S045_EK269_GAS1_v4_rep2a_virus.txt\n",
      "../Data/run1/Parsed_data//merged/S016_EK269_GAS1_p1_rep1d_plasmid-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S016_EK269_GAS1_p1_rep1d_plasmid-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S016_EK269_GAS1_p1_rep1d_plasmid-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S016_EK269_GAS1_p1_rep1d_plasmid-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S016_EK269_GAS1_p1_rep1d_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S016_EK269_GAS1_p1_rep1d_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S016_EK269_GAS1_p1_rep1d_plasmid.txt\n",
      "../Data/run1/Parsed_data//merged/S014_EK269_GAS1_p1_rep1b_plasmid-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S014_EK269_GAS1_p1_rep1b_plasmid-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S014_EK269_GAS1_p1_rep1b_plasmid-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S014_EK269_GAS1_p1_rep1b_plasmid-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S014_EK269_GAS1_p1_rep1b_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S014_EK269_GAS1_p1_rep1b_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S014_EK269_GAS1_p1_rep1b_plasmid.txt\n",
      "../Data/run1/Parsed_data//merged/S043_EK269_GAS1_v3_rep1c_virus-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S043_EK269_GAS1_v3_rep1c_virus-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S043_EK269_GAS1_v3_rep1c_virus-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S043_EK269_GAS1_v3_rep1c_virus-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S043_EK269_GAS1_v3_rep1c_virus.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S043_EK269_GAS1_v3_rep1c_virus.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S043_EK269_GAS1_v3_rep1c_virus.txt\n",
      "../Data/run1/Parsed_data//merged/S013_EK269_GAS1_p1_rep1a_plasmid-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S013_EK269_GAS1_p1_rep1a_plasmid-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S013_EK269_GAS1_p1_rep1a_plasmid-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S013_EK269_GAS1_p1_rep1a_plasmid-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S013_EK269_GAS1_p1_rep1a_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S013_EK269_GAS1_p1_rep1a_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S013_EK269_GAS1_p1_rep1a_plasmid.txt\n",
      "../Data/run1/Parsed_data//merged/S047_EK269_GAS1_v4_rep2c_virus-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S047_EK269_GAS1_v4_rep2c_virus-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S047_EK269_GAS1_v4_rep2c_virus-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S047_EK269_GAS1_v4_rep2c_virus-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S047_EK269_GAS1_v4_rep2c_virus.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S047_EK269_GAS1_v4_rep2c_virus.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S047_EK269_GAS1_v4_rep2c_virus.txt\n",
      "../Data/run1/Parsed_data//merged/S044_EK269_GAS1_v3_rep1d_virus-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S044_EK269_GAS1_v3_rep1d_virus-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S044_EK269_GAS1_v3_rep1d_virus-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S044_EK269_GAS1_v3_rep1d_virus-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S044_EK269_GAS1_v3_rep1d_virus.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S044_EK269_GAS1_v3_rep1d_virus.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S044_EK269_GAS1_v3_rep1d_virus.txt\n",
      "../Data/run1/Parsed_data//merged/S041_EK269_GAS1_v3_rep1a_virus-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S041_EK269_GAS1_v3_rep1a_virus-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S041_EK269_GAS1_v3_rep1a_virus-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S041_EK269_GAS1_v3_rep1a_virus-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S041_EK269_GAS1_v3_rep1a_virus.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S041_EK269_GAS1_v3_rep1a_virus.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S041_EK269_GAS1_v3_rep1a_virus.txt\n",
      "../Data/run1/Parsed_data//merged/S046_EK269_GAS1_v4_rep2b_virus-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S046_EK269_GAS1_v4_rep2b_virus-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S046_EK269_GAS1_v4_rep2b_virus-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S046_EK269_GAS1_v4_rep2b_virus-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S046_EK269_GAS1_v4_rep2b_virus.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S046_EK269_GAS1_v4_rep2b_virus.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S046_EK269_GAS1_v4_rep2b_virus.txt\n",
      "../Data/run1/Parsed_data//merged/S005_EK266_GAS1_p1_rep1a_plasmid-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S005_EK266_GAS1_p1_rep1a_plasmid-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S005_EK266_GAS1_p1_rep1a_plasmid-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S005_EK266_GAS1_p1_rep1a_plasmid-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S005_EK266_GAS1_p1_rep1a_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S005_EK266_GAS1_p1_rep1a_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S005_EK266_GAS1_p1_rep1a_plasmid.txt\n",
      "../Data/run1/Parsed_data//merged/S048_EK269_GAS1_v4_rep2d_virus-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S048_EK269_GAS1_v4_rep2d_virus-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S048_EK269_GAS1_v4_rep2d_virus-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S048_EK269_GAS1_v4_rep2d_virus-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S048_EK269_GAS1_v4_rep2d_virus.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S048_EK269_GAS1_v4_rep2d_virus.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S048_EK269_GAS1_v4_rep2d_virus.txt\n",
      "../Data/run1/Parsed_data//merged/S052_EK269_GAS1_v5_rep3d_virus-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S052_EK269_GAS1_v5_rep3d_virus-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S052_EK269_GAS1_v5_rep3d_virus-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S052_EK269_GAS1_v5_rep3d_virus-L4.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S052_EK269_GAS1_v5_rep3d_virus.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S052_EK269_GAS1_v5_rep3d_virus.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S052_EK269_GAS1_v5_rep3d_virus.txt\n",
      "../Data/run1/Parsed_data//merged/S006_EK266_GAS1_p1_rep1b_plasmid-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S006_EK266_GAS1_p1_rep1b_plasmid-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S006_EK266_GAS1_p1_rep1b_plasmid-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S006_EK266_GAS1_p1_rep1b_plasmid-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S006_EK266_GAS1_p1_rep1b_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S006_EK266_GAS1_p1_rep1b_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S006_EK266_GAS1_p1_rep1b_plasmid.txt\n",
      "../Data/run1/Parsed_data//merged/S050_EK269_GAS1_v5_rep3b_virus-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S050_EK269_GAS1_v5_rep3b_virus-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S050_EK269_GAS1_v5_rep3b_virus-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S050_EK269_GAS1_v5_rep3b_virus-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S050_EK269_GAS1_v5_rep3b_virus.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S050_EK269_GAS1_v5_rep3b_virus.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S050_EK269_GAS1_v5_rep3b_virus.txt\n",
      "../Data/run1/Parsed_data//merged/S049_EK269_GAS1_v5_rep3a_virus-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S049_EK269_GAS1_v5_rep3a_virus-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S049_EK269_GAS1_v5_rep3a_virus-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S049_EK269_GAS1_v5_rep3a_virus-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S049_EK269_GAS1_v5_rep3a_virus.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S049_EK269_GAS1_v5_rep3a_virus.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S049_EK269_GAS1_v5_rep3a_virus.txt\n",
      "../Data/run1/Parsed_data//merged/S051_EK269_GAS1_v5_rep3c_virus-L1.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S051_EK269_GAS1_v5_rep3c_virus-L2.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S051_EK269_GAS1_v5_rep3c_virus-L3.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//merged/S051_EK269_GAS1_v5_rep3c_virus-L4.fastq.gz\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S051_EK269_GAS1_v5_rep3c_virus.txt\n",
      "../Data/run1/Parsed_data//library/lowQ/S051_EK269_GAS1_v5_rep3c_virus.txt\n",
      "../Data/run1/Parsed_data//library/discarded/S051_EK269_GAS1_v5_rep3c_virus.txt\n"
     ]
    }
   ],
   "source": [
    "def count_reads(merge_path, output_path):\n",
    "    if merge_path[-1] is not '/': merge_path += '/'\n",
    "    if output_path[-1] is not '/': output_path += '/'\n",
    "        \n",
    "    sorted_out = output_path + 'sorted'\n",
    "    discarded_out = output_path + 'discarded'\n",
    "    lowQ_out = output_path + 'lowQ'\n",
    "    \n",
    "    check_directory_exists(sorted_out)\n",
    "    check_directory_exists(lowQ_out)\n",
    "    check_directory_exists(discarded_out)\n",
    "    \n",
    "    sample_files = glob.glob(merge_path + '*L1.fastq.gz') #use this for real data\n",
    "                \n",
    "    log = open(output_path + 'tabulate_output.txt', 'w')\n",
    "    log.write('reads\\t%good\\t%bad\\t%lowQ\\t%!primer\\tminQ\\tname\\n')\n",
    "    \n",
    "    for f in sample_files:\n",
    "        sequences = {}\n",
    "        sequences_lowQ = {}\n",
    "        sequences_discarded = {}\n",
    "        \n",
    "\n",
    "        output_file = f.split('/')[-1].split('-')[0]+'.txt' # parse out the sample name (no lane numbers)\n",
    "\n",
    "        count_finished = os.path.isfile(sorted_out + '/' + output_file)\n",
    "        if count_finished:\n",
    "            print('skipping ' + output_file)\n",
    "        else:\n",
    "            for lane_num in [1,2,3,4]:\n",
    "                input_file = f.replace('L1.', 'L{num}.'.format(num=lane_num))            \n",
    "\n",
    "                reads_processed_counts = 0\n",
    "                good_read_counts = 0\n",
    "                bad_read_counts_lowQ = 0\n",
    "                bad_read_counts_no_primer = 0\n",
    "                try:\n",
    "                    print(input_file)\n",
    "                    if \".gz\" in input_file:\n",
    "                        # print \"gz in handle\"\n",
    "                        handle = gzip.open(input_file,\"rt\") # stream the gz file\n",
    "                    else: \n",
    "                        print('handle is not gz')\n",
    "                        handle = input_file\n",
    "                    for read in SeqIO.parse(handle, 'fastq'):                        \n",
    "    #                     if reads_processed_counts >= 100 : break # stop early                            \n",
    "                        seq=str(read.seq.reverse_complement())\n",
    "                        scores = read.letter_annotations[\"phred_quality\"]\n",
    "                        scores.reverse()\n",
    "\n",
    "                        keep, library_seq, library_scores = get_libary_region(seq, scores, T21_F ,T21_R )\n",
    "\n",
    "                        if keep:\n",
    "                            min_qscore = min(library_scores)\n",
    "                            if min_qscore > MIN_QSCORE_THRESHOLD:\n",
    "                                tabulate(sequences, str(library_seq))                                \n",
    "                                good_read_counts += 1\n",
    "                            else:\n",
    "                                tabulate(sequences_lowQ, str(library_seq))\n",
    "                                bad_read_counts_lowQ += 1\n",
    "                        else:\n",
    "                            bad_read_counts_no_primer += 1\n",
    "                            tabulate(sequences_discarded, str(library_seq))\n",
    "                        reads_processed_counts += 1\n",
    "                        if(reads_processed_counts%1e6==0):\n",
    "                            print('-')\n",
    "                        elif(reads_processed_counts%1e5==0):\n",
    "                            print('.')\n",
    "                    log.write('{p}\\t{g:.2%}\\t{b:.2%}\\t{q:.2%}\\t{f:.2%}\\t{QC}\\t{n}\\n'.format(\n",
    "                        p = reads_processed_counts, \n",
    "                        g = float(good_read_counts)/max(reads_processed_counts,1),     \n",
    "                        b = float(bad_read_counts_lowQ+bad_read_counts_no_primer)/max(reads_processed_counts,1),\n",
    "                        QC = MIN_QSCORE_THRESHOLD,\n",
    "                        q = float(bad_read_counts_lowQ)/max(reads_processed_counts,1), \n",
    "                        f = float(bad_read_counts_no_primer)/max(reads_processed_counts,1),\n",
    "                        n =  input_file))        \n",
    "                except Exception as e:\n",
    "                    log.write('ERROR\\t\\t\\t\\t\\t\\t' + input_file + '\\n')\n",
    "            write_seq_counts(sequences, sorted_out + '/' + output_file)\n",
    "            write_seq_counts(sequences_lowQ, lowQ_out + '/' + output_file)\n",
    "            write_seq_counts(sequences_discarded, discarded_out + '/' + output_file)\n",
    "    log.close()             \n",
    "sample_files=count_reads(merge_path, output_path)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Combine reads from different replicas (This takes about an hour to run)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../Data/run1/Parsed_data//library/sorted/S050_EK269_GAS1_v5_rep3b_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S015_EK269_GAS1_p1_rep1c_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S051_EK269_GAS1_v5_rep3c_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S014_EK269_GAS1_p1_rep1b_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S005_EK266_GAS1_p1_rep1a_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S006_EK266_GAS1_p1_rep1b_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S044_EK269_GAS1_v3_rep1d_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S013_EK269_GAS1_p1_rep1a_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S048_EK269_GAS1_v4_rep2d_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S052_EK269_GAS1_v5_rep3d_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S016_EK269_GAS1_p1_rep1d_plasmid.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S042_EK269_GAS1_v3_rep1b_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S047_EK269_GAS1_v4_rep2c_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S049_EK269_GAS1_v5_rep3a_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S046_EK269_GAS1_v4_rep2b_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S043_EK269_GAS1_v3_rep1c_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S041_EK269_GAS1_v3_rep1a_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S045_EK269_GAS1_v4_rep2a_virus.txt\n"
     ]
    }
   ],
   "source": [
    "def combine_replicas(output_path):\n",
    "    if output_path[-1] is not '/': output_path += '/'\n",
    "    replica_files = glob.glob(output_path + 'sorted/*.txt')\n",
    "    #print(replica_files)  \n",
    "    for f in replica_files:\n",
    "\n",
    "        sequences = {}\n",
    "        output_file = f.split('.txt')[0][:]\n",
    "\n",
    "        with open(f, 'r') as fh:\n",
    "            lines = 0\n",
    "            L = fh.readline() # skip the header\n",
    "            for line in fh:\n",
    "                parts=re.split('[,\\t ]+', line.rstrip('\\n'))\n",
    "                seq = parts[0]\n",
    "                count = int(parts[1])\n",
    "                tabulate(sequences, seq, count)                                   \n",
    "                lines += 1\n",
    "#                     if lines > 10: break\n",
    "        write_seq_counts(sequences, output_file + '.txt')\n",
    "combine_replicas(output_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Build dataframe of counts for each sequence in the targeted designs. (a few minutes to run)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of sequences in target lib:  243484\n",
      "../Data/run1/Parsed_data//library/sorted/S050_EK269_GAS1_v5_rep3b_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S015_EK269_GAS1_p1_rep1c_plasmid.txt\n",
      ".\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S051_EK269_GAS1_v5_rep3c_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S014_EK269_GAS1_p1_rep1b_plasmid.txt\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S005_EK266_GAS1_p1_rep1a_plasmid.txt\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S006_EK266_GAS1_p1_rep1b_plasmid.txt\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S044_EK269_GAS1_v3_rep1d_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S013_EK269_GAS1_p1_rep1a_plasmid.txt\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S048_EK269_GAS1_v4_rep2d_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S052_EK269_GAS1_v5_rep3d_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S016_EK269_GAS1_p1_rep1d_plasmid.txt\n",
      ".\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S042_EK269_GAS1_v3_rep1b_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S047_EK269_GAS1_v4_rep2c_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S049_EK269_GAS1_v5_rep3a_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S046_EK269_GAS1_v4_rep2b_virus.txt\n",
      ".\n",
      "../Data/run1/Parsed_data//library/sorted/S043_EK269_GAS1_v3_rep1c_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S041_EK269_GAS1_v3_rep1a_virus.txt\n",
      "../Data/run1/Parsed_data//library/sorted/S045_EK269_GAS1_v4_rep2a_virus.txt\n"
     ]
    }
   ],
   "source": [
    "T21_WT = 'GACGAAGAGGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTATGGTTCTGTATCTACCAACCTCCAGAGAGGCAACAGA'\n",
    "T21_AA_WT =\"DEEEIRTTNPVATEQYGSVSTNLQRGNR\"\n",
    "def reduce_library_to_target(output_path, target_path):  \n",
    "    if output_path[-1] is not '/': output_path += '/'\n",
    "    log_path = output_path + 'target_log.txt'\n",
    "    sample_files = glob.glob(output_path + 'sorted/*.txt')\n",
    "    sample_names = list()\n",
    "    for s in sample_files:\n",
    "        name = s.split('.txt')[0].split('/')[-1]\n",
    "        name=name[name.index(\"_\")+1:]\n",
    "        sample_names.append(name)\n",
    "\n",
    "    #return sample_names\n",
    "    target = list()\n",
    "    data = {}\n",
    "\n",
    "    with open(log_path, 'w') as oh:\n",
    "        target_df = pd.read_csv(target_path)\n",
    "        lines = 0\n",
    "        nt_replicas = {}\n",
    "        aa_replicas = {}\n",
    "        target = list()\n",
    "        for i, row in target_df.iterrows():\n",
    "            aa = row[\"aa\"]\n",
    "            aa_mask = row[\"aa_mask\"]\n",
    "            nt = row[\"nt\"]\n",
    "            category = row[\"category\"]\n",
    "            control_var = row[\"control_var\"]\n",
    "            rep_var = row[\"rep_var\"]\n",
    "            chip_var = row[\"chip_var\"]\n",
    "            is_wt = 1 if nt == T21_WT else 0\n",
    "            is_aa_wt=1 if aa== T21_AA_WT else 0\n",
    "            num_mut = sum([1 if x!='_' else 0 for x in aa_mask])            \n",
    "\n",
    "            if(num_mut == 1 and len(aa_mask) == 29): # fix any instances where single insertions were not lowercase\n",
    "                for i in range(len(aa)):\n",
    "                    if(aa_mask[i]!='_'):\n",
    "                        aa_mask = aa_mask.replace(aa_mask[i], aa_mask[i].lower())\n",
    "                        aa = aa[:i] + aa_mask[i] + aa[i+1:]\n",
    "            if nt not in nt_replicas: # count the aa replicas\n",
    "                tabulate(aa_replicas, aa)\n",
    "                tabulate(nt_replicas, nt)            \n",
    "                target.append({'nt':nt,\n",
    "                               'category':category,\n",
    "                               'chip':chip_var,\n",
    "                               'control':control_var,\n",
    "                               'rep_original':rep_var,\n",
    "                               'aa':aa, \n",
    "                               'mask':aa_mask, \n",
    "                               'mut':num_mut, \n",
    "                               'is_wt_nt':is_wt, \n",
    "                               'is_wt_aa':is_aa_wt,\n",
    "                               'rep_i':aa_replicas[aa]})      \n",
    "\n",
    "        print('number of sequences in target lib: ', len(target)+1)\n",
    "        for i in range(len(target)):\n",
    "            t = target[i]\n",
    "            t['rep_total'] = aa_replicas[t['aa']]\n",
    "        # add empty counts entries to data\n",
    "        for i in range(len(target)):\n",
    "            t = target[i]\n",
    "            nt = t['nt']\n",
    "            t.pop('nt') # remove it since we already will have this info in the key\n",
    "            for n in sample_names: # set initial count for each sample to 0\n",
    "                t[n] = 0 \n",
    "            data[nt] = t    \n",
    "\n",
    "        all_total = target_total = 0\n",
    "        for s in sample_files:\n",
    "            print(s)\n",
    "            lines = all_counts = target_counts = 0\n",
    "            name = s.split('.txt')[0].split('/')[-1]\n",
    "            name=name[name.index(\"_\")+1:]\n",
    "            with open(s, 'r') as fh:            \n",
    "                fh.readline() # skip the header\n",
    "                for line in fh:\n",
    "                    parts=re.split('[,\\t ]+', line.rstrip('\\n'))\n",
    "                    seq = parts[0]\n",
    "                    count = int(parts[1])                      \n",
    "                    lines+=1\n",
    "                    all_counts += count\n",
    "                    if seq in data:\n",
    "                        data[seq][name] = count\n",
    "                        target_counts += count\n",
    "                    if lines%300000==0: print ('.')\n",
    "    #                     if lines > 100: break\n",
    "                oh.write('{name}, {L:.2E}, {N:.2E}, {T:.2E}, {percent:.3}\\n'.format(name=name, \n",
    "                                                               L=lines,\n",
    "                                                               N=all_counts, \n",
    "                                                               T=target_counts, \n",
    "                                                               percent=float(target_counts)/all_counts)) \n",
    "                all_total += all_counts\n",
    "                target_total += target_counts\n",
    "            oh.write('{name}, {L:.2E}, {N:.2E}, {T:.2E}, {percent:.3}\\n'.format(name='All', \n",
    "                                                           L=lines,\n",
    "                                                           N=all_total, \n",
    "                                                           T=target_total, \n",
    "                                                           percent=float(target_total)/all_total)) \n",
    "    return data\n",
    "        \n",
    "        \n",
    "    \n",
    "start_time = time.time()\n",
    "data = reduce_library_to_target(output_path, target_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_data = pd.DataFrame(list(data.values()), index=list(data.keys()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['EK266_GAS1_p1_rep1a_plasmid',\n",
       " 'EK266_GAS1_p1_rep1b_plasmid',\n",
       " 'EK269_GAS1_p1_rep1a_plasmid',\n",
       " 'EK269_GAS1_p1_rep1b_plasmid',\n",
       " 'EK269_GAS1_p1_rep1c_plasmid',\n",
       " 'EK269_GAS1_p1_rep1d_plasmid',\n",
       " 'EK269_GAS1_v3_rep1a_virus',\n",
       " 'EK269_GAS1_v3_rep1b_virus',\n",
       " 'EK269_GAS1_v3_rep1c_virus',\n",
       " 'EK269_GAS1_v3_rep1d_virus',\n",
       " 'EK269_GAS1_v4_rep2a_virus',\n",
       " 'EK269_GAS1_v4_rep2b_virus',\n",
       " 'EK269_GAS1_v4_rep2c_virus',\n",
       " 'EK269_GAS1_v4_rep2d_virus',\n",
       " 'EK269_GAS1_v5_rep3a_virus',\n",
       " 'EK269_GAS1_v5_rep3b_virus',\n",
       " 'EK269_GAS1_v5_rep3c_virus',\n",
       " 'EK269_GAS1_v5_rep3d_virus',\n",
       " 'aa',\n",
       " 'category',\n",
       " 'chip',\n",
       " 'control',\n",
       " 'is_wt_aa',\n",
       " 'is_wt_nt',\n",
       " 'mask',\n",
       " 'mut',\n",
       " 'rep_i',\n",
       " 'rep_original',\n",
       " 'rep_total']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(df_data.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "243483\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>EK266_GAS1_p1_rep1a_plasmid</th>\n",
       "      <th>EK266_GAS1_p1_rep1b_plasmid</th>\n",
       "      <th>EK269_GAS1_p1_rep1a_plasmid</th>\n",
       "      <th>EK269_GAS1_p1_rep1b_plasmid</th>\n",
       "      <th>EK269_GAS1_p1_rep1c_plasmid</th>\n",
       "      <th>EK269_GAS1_p1_rep1d_plasmid</th>\n",
       "      <th>EK269_GAS1_v3_rep1a_virus</th>\n",
       "      <th>EK269_GAS1_v3_rep1b_virus</th>\n",
       "      <th>EK269_GAS1_v3_rep1c_virus</th>\n",
       "      <th>EK269_GAS1_v3_rep1d_virus</th>\n",
       "      <th>...</th>\n",
       "      <th>category</th>\n",
       "      <th>chip</th>\n",
       "      <th>control</th>\n",
       "      <th>is_wt_aa</th>\n",
       "      <th>is_wt_nt</th>\n",
       "      <th>mask</th>\n",
       "      <th>mut</th>\n",
       "      <th>rep_i</th>\n",
       "      <th>rep_original</th>\n",
       "      <th>rep_total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTATGGTTCTGTATCTACCAACCTCGGGGGGGAAGGCGACAACAGA</th>\n",
       "      <td>51</td>\n",
       "      <td>44</td>\n",
       "      <td>36</td>\n",
       "      <td>58</td>\n",
       "      <td>77</td>\n",
       "      <td>37</td>\n",
       "      <td>195</td>\n",
       "      <td>227</td>\n",
       "      <td>144</td>\n",
       "      <td>168</td>\n",
       "      <td>...</td>\n",
       "      <td>rnn_designed_plus_rand_train_walked</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>__D____________________gGE_d__</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTATGGTTCTGTATCTACCAACCTCCAGGATAACGGCAACAACGATAGA</th>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "      <td>17</td>\n",
       "      <td>16</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>rnn_designed_plus_rand_train_walked</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>__D_____________________Dn_n_d_</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTATGGTGCGGTATCTACCAACCTCCAGGGCGACGGCAACGATAGA</th>\n",
       "      <td>23</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>36</td>\n",
       "      <td>47</td>\n",
       "      <td>34</td>\n",
       "      <td>50</td>\n",
       "      <td>...</td>\n",
       "      <td>rnn_designed_plus_rand_train_walked</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>__D______________A______Gd__d_</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTATGGTTCTGTATCTACCAACCTCCAGCACGACGGCGATGAAAGA</th>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>14</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>rnn_designed_plus_rand_train_walked</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>_____A__________________Hd_De_</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTATGGTAATGTATCTACCAACCTCCAGGGCGGAGGCGACAACAGA</th>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td>37</td>\n",
       "      <td>32</td>\n",
       "      <td>76</td>\n",
       "      <td>64</td>\n",
       "      <td>63</td>\n",
       "      <td>50</td>\n",
       "      <td>...</td>\n",
       "      <td>rnn_designed_plus_rand_train_walked</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>__H______________N______Gg_d__</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    EK266_GAS1_p1_rep1a_plasmid  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           51   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                            4   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           23   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...                           10   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           27   \n",
       "\n",
       "                                                    EK266_GAS1_p1_rep1b_plasmid  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           44   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                            7   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           11   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...                           16   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           20   \n",
       "\n",
       "                                                    EK269_GAS1_p1_rep1a_plasmid  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           36   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                            6   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           11   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...                           12   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           12   \n",
       "\n",
       "                                                    EK269_GAS1_p1_rep1b_plasmid  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           58   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                            8   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           12   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...                            9   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           11   \n",
       "\n",
       "                                                    EK269_GAS1_p1_rep1c_plasmid  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           77   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                            8   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           13   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...                           14   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           37   \n",
       "\n",
       "                                                    EK269_GAS1_p1_rep1d_plasmid  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           37   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                            8   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           13   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...                           11   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                           32   \n",
       "\n",
       "                                                    EK269_GAS1_v3_rep1a_virus  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                        195   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                         16   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                         36   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...                         14   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                         76   \n",
       "\n",
       "                                                    EK269_GAS1_v3_rep1b_virus  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                        227   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                         17   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                         47   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...                         12   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                         64   \n",
       "\n",
       "                                                    EK269_GAS1_v3_rep1c_virus  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                        144   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                         16   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                         34   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...                          8   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                         63   \n",
       "\n",
       "                                                    EK269_GAS1_v3_rep1d_virus  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                        168   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                         12   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                         50   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...                         12   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...                         50   \n",
       "\n",
       "                                                      ...     \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...    ...      \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...    ...      \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...    ...      \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...    ...      \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...    ...      \n",
       "\n",
       "                                                                               category  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...  rnn_designed_plus_rand_train_walked   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...  rnn_designed_plus_rand_train_walked   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...  rnn_designed_plus_rand_train_walked   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...  rnn_designed_plus_rand_train_walked   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...  rnn_designed_plus_rand_train_walked   \n",
       "\n",
       "                                                    chip  control  is_wt_aa  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...     1        0         0   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...     1        0         0   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...     1        0         0   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...     1        0         0   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...     1        0         0   \n",
       "\n",
       "                                                    is_wt_nt  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...         0   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...         0   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...         0   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...         0   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...         0   \n",
       "\n",
       "                                                                               mask  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...   __D____________________gGE_d__   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...  __D_____________________Dn_n_d_   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...   __D______________A______Gd__d_   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...   _____A__________________Hd_De_   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...   __H______________N______Gg_d__   \n",
       "\n",
       "                                                    mut  rep_i rep_original  \\\n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...    5      1            1   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...    5      1            1   \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...    5      1            1   \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...    5      1            1   \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...    5      1            1   \n",
       "\n",
       "                                                   rep_total  \n",
       "GACGAAGATGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...         1  \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...         1  \n",
       "GACGAGGACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...         1  \n",
       "GACGAAGAGGAAATCGCTACAACCAATCCCGTGGCTACGGAGCAGTA...         1  \n",
       "GACGAACACGAAATCAGGACAACCAATCCCGTGGCTACGGAGCAGTA...         1  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(df_data))\n",
    "df_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_data.to_csv('../Data/raw_counts_NextSeq_run{RUN_TO_COUNT}.txt', index=True)\n"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
